components/director/internal/domain/root_resolver.go

```diff
@@ -98,27 +99,39 @@ func NewRootResolver(
 	tokenLength int,
 	hydraURL *url.URL,
 	accessStrategyExecutorProvider *accessstrategy.Provider,
-	cache certloader.Cache,
-) *RootResolver {
+) (*RootResolver, error) {
 	oAuth20HTTPClient := &http.Client{
 		Timeout:   oAuth20Cfg.HTTPClientTimeout,
 		Transport: httputil.NewCorrelationIDTransport(httputil.NewServiceAccountTokenTransport(http.DefaultTransport)),
 	}
 
-	oauthCredentials := &authpkg.OAuthCredentials{
-		ClientID:     selfRegConfig.ClientID,
-		ClientSecret: selfRegConfig.ClientSecret,
-		TokenURL:     selfRegConfig.URL + selfRegConfig.OauthTokenPath,
+	var credentials authpkg.Credentials
+	if selfRegConfig.OAuthMode == oauth.Standard {
+		credentials = &authpkg.OAuthCredentials{
+			ClientID:     selfRegConfig.ClientID,
+			ClientSecret: selfRegConfig.ClientSecret,
+			TokenURL:     selfRegConfig.TokenURL + selfRegConfig.OauthTokenPath,
```

Comment:
I think if we have 'standard' mode the TokenURL will be the same as URL so the functionality will be the same. Also, I think it is better to use TokenURL because we are assigning it to TokenURL field.

Reply:
A couple of things:
1. the only mode I see in the charts(local and on productive setup) is `oauth-mtls` 
2. I didn't see the urls(URL and TokenURL to be populated based on oauth mode? They get their values independently from the oauth mode or I may miss something?
3. URL field and TokenURL field are populated through different properties. In  `compass/chart/compass/charts/director/templates/deployment.yaml` file `APP_SELF_REGISTER_URL` env is populated through secret with key `global.director.selfRegister.secret.urlKey` and `APP_SELF_REGISTER_TOKEN_URL` from secret with key `global.director.selfRegister.secret.tokenUrlKey`. Locally and on productive setup both of them have different values? Am I missing something?

Reply:
Overall that's the credentials used "caller config" for token request, I think the URL have different semantic but I might be wrong. What are the difference between URL and TokenURL, when we need to use the former and when the other one? Do we need both of them if they are same in `standard` oauth mode? Are they different in `oauth-mtls` mode?

Reply:
The difference is that we call URL for self-registration and TOKEN_URL for getting the token.

Reply:
Yes, the oauth-mtls is the only OAuth mode that we have(local and on productive setup) and this part of the code is left only for backward compatibility. Also, the URLs are not populated based on OAuth mode. I will revert this change to use URL instead of TokenURL.